packages/expect/src/__tests__/extend.test.ts

```diff
@@ -229,3 +230,18 @@ it('throws descriptive errors for invalid matchers', () => {
     'expect.extend: `default` is not a valid matcher. Must be a function, is "string"',
   );
 });
+
+it('exposes matchers in context', () => {
+  jestExpect.extend({
+    shouldNotError(_actual: unknown, _expected: unknown) {
+      const pass = this.equals(this.matchers, matchers);
```

Comment:
Hm.. I still see type error here:

```suggestion
    shouldNotError(_actual: unknown) {
      const pass: boolean = this.equals(this.matchers, matchers);
```